config.go

```diff
@@ -243,6 +244,10 @@ url = "http://localhost:8086" # required.
 # The target database for metrics. This database must already exist
 database = "telegraf" # required.
 
+# Timeout in seconds (for the connection with InfluxDB).
+# If not provided, will default to 0 (no timeout)
+# timeout = 5
```

Comment:
The convention in other InfluxDB products is to make this a string, and parse it as a `toml.Duration` type. This allows the timeout to be set as, say, "5s" or "100ms". Check the InfluxDB source for examples:

https://github.com/influxdb/influxdb/blob/master/tsdb/config.go